lib/src/logger.dart

```diff
@@ -28,12 +30,32 @@ abstract class Logger {
   /// a deprecation warning. Implementations should surface all this information
   /// to the end user.
   void warn(String message,
-      {FileSpan? span, Trace? trace, bool deprecation = false});
+      {FileSpan? span,
+      Trace? trace,
+      @Deprecated("Use `warnForDeprecation` instead.") bool deprecation =
+          false});
 
   /// Emits a debugging message associated with the given [span].
   void debug(String message, SourceSpan span);
 }
 
+/// An extension to add a `warnForDeprecation` method to loggers without
+/// making a breaking API change.
+///
+/// {@category Compile}
+extension WarnForDeprecation on Logger {
+  /// Emits a deprecation warning for [deprecation] with the given [message].
+  void warnForDeprecation(Deprecation deprecation, String message,
+      {FileSpan? span, Trace? trace}) {
+    var self = this;
+    if (self is DeprecationHandlingLogger) {
+      self.handleDeprecationWarning(deprecation, message, span, trace);
```

Comment:
It might be clearer to call the method on `DeprecationHandlingLogger` `warnForDeprecation()`. The call won't be ambiguous here because the static type of `self` in this branch is `DeprecationHandlingLogger`.

Reply:
Done